File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer=1
    if len(numbers)==1:
        if numbers[0]==target or numbers[0] == -1*target:
            return 1
        else:
            return 0
    else:
        temp=numbers[:]
        a=temp.pop()
        return solution(temp, target+a)+solution(temp, target-a)
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    vector<int>array(numbers.size());
    if(numbers.size()==1){
        if(numbers[0]==target || numbers[0] == -1*target)
            answer=1;
        else
            answer=0;
    }
    else{
        int a=numbers[numbers.size()-1];
        numbers.pop_back();
        answer=solution(numbers, target+a)+solution(numbers, target-a);
    }
    return answer;
}